jetty-server/src/main/java/org/eclipse/jetty/server/HostHeaderCustomizer.java

```diff
@@ -55,14 +58,29 @@ public HostHeaderCustomizer(String serverName)
      */
     public HostHeaderCustomizer(String serverName, int serverPort)
     {
-        this.serverName = Objects.requireNonNull(serverName);
-        this.serverPort = serverPort;
+        String host = Objects.requireNonNull(serverName);
+        if (serverPort > HttpURI.NO_PORT)
+            host += ":" + serverPort;
+        hostValue = host;
     }
 
     @Override
     public void customize(Connector connector, HttpConfiguration channelConfig, Request request)
     {
-        if (request.getHeader("Host") == null)
-            request.setAuthority(serverName, serverPort);  // TODO set the field as well?
+        String hostHeaderValue = request.getHeader("Host");
+
+        // No Host Header
+        if (request.getHttpVersion().getVersion() != HttpVersion.HTTP_1_1.getVersion() &&
```

Comment:
BTW, I see IP gets leaked even for HTTP/1.1 when the Host header has single and/or multiple whitespace values like the following (see cookie value
```
curl  http://localhost:6109  -vl --http1.1 --Header "Host:  "
* Rebuilt URL to: http://localhost:6109/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 6109 (#0)
> GET / HTTP/1.1
> Host: 
> User-Agent: curl/7.58.0
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Tue, 14 Dec 2021 01:09:48 GMT
< Set-Cookie: CookieConsentPolicy=0:0; domain=127.0.0.1; path=/; expires=Wed, 14-Dec-2022 01:09:49 GMT; Max-Age=31536000
< Set-Cookie: LSKey-c$CookieConsentPolicy=0:0; domain=127.0.0.1; path=/; expires=Wed, 14-Dec-2022 01:09:49 GMT; Max-Age=31536000
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block

```
If we have a null host then we get BadMessage as expected for HTTP/1.1, but for an empty host, we see IP. Do we need else case here to check the host with white spaces?